src/trie.c

```diff
@@ -84,3 +81,83 @@ int insert_string(char *word, trie_t *t)
         return insert_string(word, t->children[(unsigned)curr]);
     }
 }
+
+
+trie_t *trie_search_end(char* word, trie_t *t)
+{
+
+    int len;
+    trie_t* curr;
+    trie_t** next;
+
+    len = strlen(word);
+    curr = t;
+    next = t->children;
+
+    for (int i=0; i<len; i++) {
+        int j = (int) word[i];
+        curr = next[j];
+
+        if (curr == NULL)
+            return NULL;
+
+        next = next[j]->children;
+    }
+
+    return curr;
+}
+
+
+int trie_search(char* word, trie_t *t)
+{
+    trie_t *end = trie_search_end(word,t);
+
+    if (end == NULL)
+        return NOT_IN_TRIE;
+
+    if (end->is_word == 1) 
+        return IN_TRIE;
+
+    return PARTIAL_IN_TRIE;
+}
+
+int count_completion_recursive( trie_t *t){
+
+	int acc = 0;
+
+	if (t == NULL)
+		return acc;
+
+	if (t->is_word == 1)
+		acc++;
+
+	for (int i=0;i<256;i++)
+		acc += count_completion_recursive(t->children[i]);
+
+	return acc;
+}
+
+/*
```

Comment:
This is the only function whose header comment is duplicated from the header file. Please be consistent.